state-transition/core/state_processor_staking.go

```diff
@@ -157,6 +158,13 @@ func (sp *StateProcessor[
 	// Get the current epoch.
 	epoch := sp.cs.SlotToEpoch(slot)
 
+	// Verify that the deposit has the ETH1 withdrawal credentials.
+	if dep.GetWithdrawalCredentials()[0] != types.EthSecp256k1CredentialPrefix {
+		return errors.New(
+			"deposit does not have ETH1 withdrawal credentials",
+		)
+	}
```

Comment:
not sure we should err here. Similarly to what we do with faulty signature, I think this deposit should be skipped and the others be duly processed

Reply:
done